### PR TITLE
doppler: add version test & shell completion

### PR DESCRIPTION
**Pull request by superherointj**
doppler:
- add version test
- add shell completion
- add flag
- minor clean-up

### CODE REVIEW

**Comment by superherointj** [comment]
Result of `nixpkgs-review pr 189783` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>doppler</li>
  </ul>
</details>

File: pkgs/tools/security/doppler/default.nix [hunk 1]
```diff
@@ -13,15 +19,29 @@ buildGoModule rec {
...
+
+  nativeBuildInputs = [ installShellFiles ];
 
   postInstall = ''
-    mv $out/bin/cli $out/bin/doppler
+    mv $out/bin/cli $out/bin/${pname}
+    installShellCompletion --cmd ${pname} \
+      --bash <($out/bin/${pname} completion bash) \
+      --fish <($out/bin/${pname} completion fish) \
+      --zsh <($out/bin/${pname} completion zsh)
```
Comment (SuperSandro2000) [inline pkgs/tools/security/doppler/default.nix:34]:
```suggestion
    mv $out/bin/cli $out/bin/doppler
    installShellCompletion --cmd doppler \
      --bash <($out/bin/doppler completion bash) \
      --fish <($out/bin/doppler completion fish) \
      --zsh <($out/bin/doppler completion zsh)
```

Comment (superherointj) [reply pkgs/tools/security/doppler/default.nix]: Done

**Comment by superherointj** [comment]
Thanks!

**Pull request merged**